### PR TITLE
Task/eparker71/tlt 1873   add guidance text to analytics

### PR DESCRIPTION
**Pull request by eparker71**
This PR adds a text block to the course analytics page in Canvas. 
The related story is TLT-1873:
[
Add a note on the Analytics dashboard in Canvas that says:

"The Faculty Oversight Committee on the Access to Electronic Information provides <link>guidance to faculty</link> on using these analytics."

link goes to this Harvard Guide page: https://wiki.harvard.edu/confluence/display/canvas/Electronic+Information+in+Canvas
]

### CODE REVIEW

File: js/analytics_text.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+
+function initAnalyticsTextBlock(){
+
+	MutationObserver = window.MutationObserver || window.WebKitMutationObserver;
+	var analyticsObserver = new MutationObserver(function (mutations) {
+		mutations.forEach(function (mutation) {
+			// check to see if the text block has already been added
+			if( $('.center-alert-info').length == 0) {
```
Comment (elliottyates) [inline js/analytics_text.js]: I would use an ID for this, so you're sure it's unique -- there could theoretically be other things on the page with that class, as it's a style-related class, not a behavior-related class.

File: js/analytics_text.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+				var $target = $(mutation.target);
+				// if the course_graphs class exists, the analytics have been loaded
+				// so go ahead and add the text block
+				if ($target.find('.course_graphs').length != 0) {
+					addAnalyticsTextBlock();
```
Comment (elliottyates) [inline js/analytics_text.js:35]: maybe remove the mutation observer at this point?

Comment (elliottyates) [inline js/analytics_text.js:35]: (We don't do this in the other mutation code we've written, and maybe there's a reason, but just checking)

**Comment by elliottyates** [comment]
As mentioned in PR #8, it would be nice to fix up the whitespace. Looks good. :+1:

**Comment by elliottyates** [comment]
Sorry to not be more specific -- I think the classes themselves were fine, because they are inherently generic / style-related. But maybe add an ID to the container, and use that ID to check for existence on the page (bonus: it's slightly faster than checking for classes with jQuery). Maybe Vittorio can weigh in on that too.

File: js/analytics_text.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+
+function initAnalyticsTextBlock(){
+
+	MutationObserver = window.MutationObserver || window.WebKitMutationObserver;
+	var analyticsObserver = new MutationObserver(function (mutations) {
+		mutations.forEach(function (mutation) {
+			// check to see if the text block has already been added
+			if( $('#center-alert-info').length == 0) {
```
Comment (elliottyates) [inline js/analytics_text.js]: (i.e. check for `#analytics-guidance` instead of a class, here)

File: js/analytics_text.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+
+function addAnalyticsTextBlock() {
+	// get the name of the current page
+	var p = window.location.pathname.split("/");
+	var resource = p[p.length - 1];
+
+	// if we are on the analytics page display the analytics text block
+	if (resource == "analytics") {
+		var guidance_text = '<div id="center-alert-info-id">' +
```
Comment (elliottyates) [inline js/analytics_text.js]: (and leave `center-alert-info` as a class, but add `id="analytics-guidance"` to the div as well)

File: css/analytics_text.css [hunk 5]
```diff
@@ -0,0 +1,42 @@
+
+/* CSS for text block on analytics page */
+
+#center-alert-info-id {
```
Comment (elliottyates) [inline css/analytics_text.css]: (and switch these back to classes)

File: js/analytics_text.js (hunk 2 above)
Comment (eparker71) [inline js/analytics_text.js:35]: done analyticsObserver.disconnect();

**Comment by eparker71** [comment]
white space is gone

**Pull request merged**